src/pybind/cephfs/cephfs.pyx

```diff
@@ -1901,6 +1901,23 @@ cdef class LibCephFS(object):
                           st_mtime=datetime.fromtimestamp(stx.stx_mtime.tv_sec),
                           st_ctime=datetime.fromtimestamp(stx.stx_ctime.tv_sec))
 
+    def is_dir_exists(self, path):
+        try:
+            return (self.stat(os.path.join(b'', path)).st_mode &
+                    0o0040000) != 0
+        except ObjectNotFound:
+            return False
+
+    def is_file_exists(self, path):
+        try:
+            return (self.stat(os.path.join(b'', path)).st_mode &
+                    0o0040000) == 0
```

Comment:
Or here it should be `s/0o0040000/0o0100000/` for file ?

Reply:
removed.